release notes: v12 preview (#770)
mia-platform/documentation: docs/release-notes/v12.0.0.md

#### Bug Fix

This version addressed few bugs, here a comprehensive list:

* solved an issue that causes to regenerate indexes and information of Projection Changes collections of Real-Time Updater attached to System of Record
* solved an issue that caused the No Code ER Schema to stop working if the JSON was completely empty
* now it's not possible to create Projections and Single Views with names longer than 80 characters, complying with the MongoDB CRUD requirements

### Fast Data Services

#### One Fast Data Service (e.g. Real Time Updater)

The new version `vX.Y.Z` of the _put here the fast data service name_ is available!

##### New Features

* put here
* the list
* of all the new features

#### Improvements

* put here
* the list
* of all the improvements

#### Bug Fix

This version addressed few bugs, here a comprehensive list:

* list here all the bug fix
* but do not highlight them so much
* just put a short description 

For more information check out the service changelog (add link to the changelog)

> ## Microfrontend Composer

Reviewer: ```suggestion
## Backoffice becomes Microfrontend Composer

:::info
Backoffice has a brand new name: Microfrontend Composer! From now on, for all the features related to the formerly known Backoffice Configurator, you will find references to Microfrontend Composer. The [documentation](/microfrontend-composer/what-is.md) has been updated accordingly.
:::
```

### New Features

#### Put here the title of the feature

Put here the description of the feature. Remember to add the documentation link

![add here your picture if needed and give it a name](./img/path/to/your/picture.png)

### Improvements

#### Put here the title of the improvement

Put here the description of the improvement

### Bug Fix

This version addressed few bugs, here a comprehensive list:

* list here all the bug fix
* but do not highlight them so much
* just put a short description 

## Back-Kit Library

> The new version `v1.4.7` of the Backoffice library (back-kit) is available!

Reviewer: ```suggestion
The new version `v1.4.7` of the Back-kit library is available!
```

:::info
To learn more about **GitOps-based deployment**, check out the dedicated [documentation page](/development_suite/deploy/gitops-based/index.md)
:::

#### Listeners management

From now on you can choose to **expose your endpoints on different Listeners** to map them to different exposed domains. In the dedicated section of the Design area, you can map each Listener to a single port in order to route external client requests towards resources exposed on your cluster. You can also see a general overview of which endpoints are exposed on which listeners, thanks to the dedicated Endpoints tab.
:::info
To learn more about Listener management, visit the dedicated [documentation page](development_suite/api-console/api-design/listeners.md)
:::
:::caution
At the moment, Listeners are only available for Projects using an **Envoy API Gateway**.
:::

## Fast Data

### Fast Data Configurator

#### Import and Export of Fast Data resources between projects

> 

Reviewer: @fedepini aggiungiamo un link alla doc?

Reviewer 2: @TheRusCompass fatto

:::

#### Listeners management

From now on you can choose to **expose your endpoints on different Listeners** to map them to different exposed domains. In the dedicated section of the Design area, you can map each Listener to a single port in order to route external client requests towards resources exposed on your cluster. You can also see a general overview of which endpoints are exposed on which listeners, thanks to the dedicated Endpoints tab.
:::info
To learn more about Listener management, visit the dedicated [documentation page](development_suite/api-console/api-design/listeners.md)
:::
:::caution
At the moment, Listeners are only available for Projects using an **Envoy API Gateway**.
:::

## Fast Data

### Fast Data Configurator

#### Unique values for ingestion topics and pr-update topics

To further improve the automatic configuration of the Fast Data, from now on when configuring a Projection inside a System of Record it is possible to set only one _ingestion topic_ and only one _pr-update topic_, instead of a topic for each environment.

> Every new projection created will automatically set the ingestion topic to the pattern `{tenantId}.{projectId}.{systemId}.{projectionId}.ingestion` and the pr-update topic with `{tenantId}.{projectId}.{systemId}.{projectionId}.pr-update`. These are editable and it is possible to use environment variables for the entire name or part of the name.

Reviewer: ```suggestion
Every new projection created will automatically set the ingestion topic to the pattern `{tenantId}.{projectId}.{systemId}.{projectionId}.ingestion` and the pr-update topic with `{tenantId}.{projectId}.{systemId}.{projectionId}.pr-update`. These are editable and it is possible to use environment variables for the entire name or part of the name. This lets you differentiate topics for each environment leveraging on the usage of environment variables.
```

These variables are still available as set as editable. They can be deleted after verifying that they are not used any longer.
:::

#### Single View Trigger Generator services attachable to Single Views

The Single View Creator detail page, used to configure the config maps of the Single View Creator service, now includes a new tab that allow to attach also a [Single View Trigger Generator] to the service, and therefore to the Single View.

![Single View Trigger Generator page](../fast_data/configuration/img/svtg-configuration.png)

Attaching a Single View Trigger Generator will simplify the configuration of the service itself by automatically setting the same ER Schema config map used in the Single View Creator, and including a dedicated space where to configure the [Kafka Projection Updates](/fast_data/configuration/config_maps/kafka_projection_updates.md) and the [Projection Changes Schema](/fast_data/configuration/config_maps/projection_changes_schema.md) config map.

:::info
This configuration cannot be used in case the Kafka Projection Updates uses manual strategies
:::

#### Upsert/Delete Strategy configuration page

> feat:

Reviewer: ```suggestion
```

The Single View Creator detail pag includes another tab, allowing the user to simplify the configuration of the _Upsert strategy and the Delete strategy_:

![Upsert/Delete Strategy page](../fast_data/configuration/img/upsert-delete-strategy.png)

Any update here will be reflected to the Single View Creator service after saving the configuration. For more information, [read more in the dedicated documentation page](/fast_data/configuration/single_view_creator.md#upsert-delete-strategies). 

> #### New Features
> 
> ##### Put here the title of the feature
> 
> Put here the description of the feature. Remember to add the documentation link
> 
> ![add here your picture if needed and give it a name](./img/path/to/your/picture.png)
> 
> #### Improvements
> 
> ##### Put here the title of the improvement
> 
> Put here the description of the improvement
> 

Reviewer: ```suggestion
```

![Upsert/Delete Strategy page](../fast_data/configuration/img/upsert-delete-strategy.png)

Any update here will be reflected to the Single View Creator service after saving the configuration. For more information, [read more in the dedicated documentation page](/fast_data/configuration/single_view_creator.md#upsert-delete-strategies). 

#### Bug Fix

This version addressed few bugs, here a comprehensive list:

* solved an issue that causes to regenerate indexes and information of Projection Changes collections of Real-Time Updater attached to System of Record
* solved an issue that caused the No Code ER Schema to stop working if the JSON was completely empty
* now it's not possible to create Projections and Single Views with names longer than 80 characters, complying with the MongoDB CRUD requirements

### Fast Data Services

#### Projection Storer

The version `v1.0.0` of the _Projection Store_ is now available!

This is a new service that can be employed in Fast Data _event-driven_ architecture in place of
Real-Time Updater plugin.  

> Similarly to Real-Time Updater plugin, Projection Storer consumes change events as ingestion messages,
> transform them into projection records, filtering and validating each of their fields and eventually save them onto the specified storage system, emitting a proper notification for downstream components.

Reviewer: ```suggestion
Similarly to Real-Time Updater plugin, Projection Storer consumes change events as ingestion messages,
transforms them into projection records, filters and validates each of their fields and eventually saves them onto the specified storage system, emitting a proper notification for downstream components.
```

- Component `bk-layout` correctly renders in `leftMenu` mode

## Marketplace

### Marketplace Updates

#### CRUD Service

The new version `vX.Y.Z` of the CRUD Service is available!

##### New Features

###### Put here the title of the feature

Put here the description of the feature. Remember to add the documentation link

##### Improvements

###### Put here the title of the improvement

Put here the description of the improvement

##### Bug Fix

This version addressed few bugs, here a comprehensive list:

* list here all the bug fix
* but do not highlight them so much
* just put a short description

#### Flow Manager Service

The new version `v2.6.0` of the Flow Manager Service is available!

##### New Features

###### Added support for side effects fired alongside commands

For each state of the machine it is now possible to specify an array of side effects command to file alongside the main command in a fire-and-forget manner.

To learn more about this feature, check out the dedicated [documentation page](/runtime_suite/flow-manager-service/30_configuration.md#states-of-the-machine)

###### Added hook functions to build commands and side effects custom payload

> Commands payload can now be generated thought a custom function that takes as input the entire Saga entity and the message label of the command, and returns as output an object with the payload to use.

Reviewer: ```suggestion
Commands payload can now be generated through a custom function that takes as input the entire Saga entity and the message label of the command, and returns as output an object with the payload to use.
```
